### PR TITLE
Issue #10020: Fix Annotation array initalization issue with curlies  on seperate line

### LINKED ISSUE #10020: annotation array initialization indentation issue
We upgraded Checkstyle from version 8.29 to 8.42, and we're seeing a strange violation around the indentation of annotation arrays.

Example:                                                                                                                                                                                                             
 
```
PS C:\abl\checkstyle-issue> type checkstyle-config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">

<module name="Checker">
    <module name="TreeWalker">
        <module name="Indentation"/>
    </module>
</module>

PS C:\abl\checkstyle-issue> type Story.java

/**
 * A class detailing fields common to all Story types.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY)
@JsonSubTypes(
    {
        @JsonSubTypes.Type(value = NonFictionStory.class, name = "NON_FICTION"),
        @JsonSubTypes.Type(value = MysteryStory.class, name = "MYSTERY"),
        @JsonSubTypes.Type(value = ScienceFictionStory.class, name = "SCIENCE_FICTION"),
        @JsonSubTypes.Type(value = HorrorStory.class, name = "HORROR"),
    }
)
public class Story {
}

PS C:\abl\checkstyle-issue> java -jar checkstyle-8.42-all.jar -c checkstyle-config.xml Story.java
Starting audit...
[ERROR] C:\abl\checkstyle-issue\Story.java:9:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] C:\abl\checkstyle-issue\Story.java:10:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] C:\abl\checkstyle-issue\Story.java:11:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] C:\abl\checkstyle-issue\Story.java:12:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 4 errors.
```
To clear the errors, I have to change the indentation of each @JsonSubTypes.Type in the array to be at the same indentation level as the curly brackets on lines 8 and 13. My understanding is using the default indentation value of 4, each array item should be indented 4 spaces from the curly brackets.

This behavior appears to have been introduced on version 8.40, if I downgrade Checkstyle to version 8.39, I don't have this problem.

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
Fixes #10020 
Diff Regression projects: https://gist.githubusercontent.com/aryaniiit002/97577199a93846562de23a0d6304e496/raw/eb4a3a70f46a644ce06914ac5ba8fd1b0e3c940b/projects-to-test-on.properties
Diff Regression config: https://gist.githubusercontent.com/Abhishek-kumar09/0745ecf11934e0214b9187f9d4896e4a/raw/2143ba7f21dcafb998019a1ca2c0ee3f1c510646/my_check.xml
Report label: Default Indentation profile

### CODE REVIEW

**Comment by Abhishek-kumar09** [comment]
Github, generate report

**Comment by Abhishek-kumar09** [comment]
How can I write indentation testcases without trailing comments like `indent:x expected:x`?

**Comment by Abhishek-kumar09** [comment]
Github, generate report

**Comment by romani** [comment]
> How can I write indentation testcases without trailing comments like indent:x expected:x?

why do you need this ?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AnnotationArrayInitHandler.java [hunk 1]
```diff
@@ -96,15 +96,24 @@ protected IndentLevel getChildrenExpectedIndent() {
...
+        // the code block should be aligned to lcurly pos + lineWrappingIndent
+        if (firstChildPos == NOT_EXIST && lcurlyPos == getLineStart(leftCurly)) {
+            expectedIndent = IndentLevel.addAcceptable(expectedIndent,
+                        lcurlyPos + getLineWrappingIndentation());
+        }
+
         if (firstChildPos != NOT_EXIST) {
             expectedIndent = IndentLevel.addAcceptable(expectedIndent, firstChildPos, lcurlyPos
-                    + getLineWrappingIndentation());
+                + getLineWrappingIndentation());
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AnnotationArrayInitHandler.java:114]: revert or make it in separate commit as "minor: formatting":
```
expectedIndent = IndentLevel.addAcceptable(expectedIndent, firstChildPos,
       lcurlyPos + getLineWrappingIndentation())
```

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java [hunk 2]
```diff
@@ -601,6 +601,19 @@ public void testOddAnnotations()
         verifyWarns(checkConfig, fileName, expected);
     }
 
+    @Test
+    public void testAnnotationOddStyles() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(IndentationCheck.class);
+
+        checkConfig.addAttribute("tabWidth", "8");
```
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java:608]: please use "addProperty", just merged depreacation of addAttribute

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnnotationArrayInitOldStyle.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/* Config:                                                                  //indent:0 exp:0
+ * This test-input is intended to be checked using following configuration: //indent:1 exp:1
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnnotationArrayInitOldStyle.java:2]: make it:
```
/*
Indentation
basicOffset = 4 
....
```

it is new format based on GSOC 21 project

**Review by romani — CHANGES_REQUESTED** [review]
items:

**Comment by Abhishek-kumar09** [comment]
> > How can I write indentation testcases without trailing comments like indent:x expected:x?
> 
> why do you need this ?

Because `//` are inferred as non blank character that changes the result of 
https://github.com/checkstyle/checkstyle/blob/175c5ede6f94185077ed934f8debea451322bca5/src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AnnotationArrayInitHandler.java#L101-L102

That ultimately affects the log error.

e.g.
## case1: (ran against default indentation profile)
```
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY)
@JsonSubTypes(
    { 
        @JsonSubTypes.Type(value = NonFictionStory.class, name = "NON_FICTION"),
        @JsonSubTypes.Type(value = MysteryStory.class, name = "MYSTERY"),
        @JsonSubTypes.Type(value = ScienceFictionStory.class, name = "SCIENCE_FICTION"),
        @JsonSubTypes.Type(value = HorrorStory.class, name = "HORROR"),
    }
)
public class Story {
}
```

output: 
```
Starting audit...
[ERROR] /home/abhishhek/playground/playground/saves.java:5:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] /home/abhishhek/playground/playground/saves.java:6:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] /home/abhishhek/playground/playground/saves.java:7:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
[ERROR] /home/abhishhek/playground/playground/saves.java:8:9: 'annotation array initialization' child has incorrect indentation level 8, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 4 errors.
```
## case 2: (ran against default indentation profile)
```diff
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY)
@JsonSubTypes(
-    { 
+   {   // <-----------
        @JsonSubTypes.Type(value = NonFictionStory.class, name = "NON_FICTION"),
        @JsonSubTypes.Type(value = MysteryStory.class, name = "MYSTERY"),
        @JsonSubTypes.Type(value = ScienceFictionStory.class, name = "SCIENCE_FICTION"),
        @JsonSubTypes.Type(value = HorrorStory.class, name = "HORROR"),
    }
)
public class Story {
}
```
output:
```
Starting audit...
Audit done.
```

### This issue is now resolved.

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java (hunk 2 above)
Comment (Abhishek-kumar09) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java]: Done

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnnotationArrayInitOldStyle.java (hunk 3 above)
Comment (Abhishek-kumar09) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationAnnotationArrayInitOldStyle.java]: Done

**Comment by romani** [comment]
CI failure:
`Caused by: org.eclipse.aether.resolution.ArtifactResolutionException: Could not find artifact org.xwiki.commons:xwiki-commons-pom:pom:13.6-SNAPSHOT`

**Comment by romani** [comment]
Github, rebase

**Comment by Abhishek-kumar09** [comment]
All checks passed.

**Review by strkkk — APPROVED** [review]

**Review by romani — APPROVED** [review]

**Pull request merged**